firebase-perf/src/main/java/com/google/firebase/perf/session/PerfSession.java

```diff
@@ -179,11 +157,10 @@ public static com.google.firebase.perf.v1.PerfSession[] buildAndSort(
   }
 
   /** If true, Session Gauge collection is enabled. */
-  public static boolean shouldCollectGaugesAndEvents() {
+  public static boolean shouldCollectGaugesAndEvents(String sessionId) {
     ConfigResolver configResolver = ConfigResolver.getInstance();
-
     return configResolver.isPerformanceMonitoringEnabled()
-        && Math.random() < configResolver.getSessionsSamplingRate();
+        && (Math.abs(sessionId.hashCode() % 100) < configResolver.getSessionsSamplingRate() * 100);
```

Comment:
This is really cool.

Do we have unit tests to ensure it's consistent for the same string, barring config flags?

Reply:
we have unit tests already to test this with RC override